### ISSUE TITLE
Root UIViewcontroller freezes

### ISSUE THREAD

**Issue by zaikshev88** [mikefrederick/MFSideMenu#33]
Hi, I am using the MFSideMenu together with a UITabbarcontroller.

My code as follows:

```
//Setting up of tab bar controllers
NSMutableArray *controllers = [NSMutableArray new];
[controllers addObject:[self sideMenu:feedNavigationController].navigationController];
[controllers addObject:[self sideMenu:popularNavigationController].navigationController];
[controllers addObject:[self sideMenu:questionsNavigationViewController].navigationController];
[controllers addObject:[self sideMenu:notificationNavigationViewController].navigationController];
[controllers addObject:[self sideMenu:profileNavigationViewController].navigationController];

[self.tabController setViewControllers:controllers];
```

Code to initiate sideMenu:

```
- (MFSideMenu *)sideMenu: (DMNavigationController*) navigationController {
    DMSideMenuViewController *sideMenuController = [[DMSideMenuViewController alloc] init];

    MFSideMenuOptions options = MFSideMenuOptionShadowEnabled|MFSideMenuOptionMenuButtonEnabled|MFSideMenuOptionBackButtonEnabled;
    MFSideMenuPanMode panMode = MFSideMenuPanModeNavigationBar|MFSideMenuPanModeNavigationController;

    MFSideMenu *sideMenu = [MFSideMenu menuWithNavigationController:navigationController
                                             sideMenuController:sideMenuController
                                                       location:MFSideMenuLocationLeft
                                                        options:options
                                                        panMode:panMode];

    sideMenuController.sideMenu = sideMenu;

    return sideMenu;
}
```

In my sideMenuController, I have a didSelectRow event that adds a new controller on top of the current stack:

```
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    if (indexPath.row == 0) {

        DMUserProfileViewController *upvc = [[DMUserProfileViewController alloc]initWithNickname:[UserInfo sharedUserInfo].getNickname];

        [self.sideMenu.navigationController pushViewController:upvc animated:NO];
        [self.sideMenu setMenuState:MFSideMenuStateHidden];
    }
}
```

The issue is that when I click on the tab to 'pop' to root view (after loading a new viewcontroller), the rootviewcontroller freezes i.e. I am unable to interact with the page at all e.g. scroll or click on cell etc

Is there any reason to cause the rootviewcontroller to freeze up?

Thanks and regards

Zhen

**Comment by zaikshev88** [comment]
I did a NSLog and it seems that the root view controller is handling some gestures which it should not be
![Screen Shot 2012-12-14 at 12 56 08 PM](https://f.cloud.github.com/assets/808970/12640/be0d1de8-45aa-11e2-99a8-96fc737df29b.png)

It seems like when I try to scroll the view, it will call this method
- (void) handleNavigationBarPan:(UIPanGestureRecognizer *)recognizer

and trigger the methods
[self pointAdjustedForInterfaceOrientation:panGestureOrigin]; and 
[self setRootControllerOffset:0];

Is there anyway to prevent these from being invoked after I pop all viewcontrollers from the stack?

**Comment by jackysze** [comment]
I've got a similar problem with the library. I'm using MFSideMenu with UIViewController + UITableView.

On the side menu, I have 3 items and 2 of them will show view controllers with UITableView and the other one will show a modal UIViewController. When a user trigger the modal view and dismiss afterwards, the app will go back to one of the UIViewControllers with UITableView. However, the UITableViews will be frozen and not responding to any touches after dismissing the modal view unless I open the side menu again.

**Comment by zaikshev88** [comment]
Hi @jackysze, I am not sure if our issues are due to the same cause, but I did find out the culprit that causes the 'freezing' for my case

In MFSideMenu.m file, look for this method

```
toggleSideMenuHidden:(BOOL)hidden
```

There is a line that sets the UserInteraction

```
self.navigationController.topViewController.view.userInteractionEnabled = (self.menuState == MFSideMenuStateHidden);
```

You can either tweak that or remove that line to solve the problem

Thanks

**Comment by jackysze** [comment]
Hi @zaikshev88 thanks for your suggestion. I've never thought of the user interaction flag has been disabled. Everything works smoothly now. You are my savior. :)

**Comment by mikefrederick** [comment]
Hmm.. glad you guys found the culprit, from the sounds of it this is a bug. Somehow the menu thinks it's open when it is closed.

**Comment by gonecoding** [comment]
When pushing a new view controller to the navigation controller's stack and hiding the menu afterwards the user interaction flag is being enabled for the newly pushed view controller (where it never has been disabled) instead of the previously top-most view controller (where it actually has been disabled before):

```
self.navigationController.topViewController.view.userInteractionEnabled = (self.menuState == MFSideMenuStateHidden);
```

That's why the previously top-most view controller's view remains disabled.

I fixed that behavior by setting the user interaction flag on the whole stack of view controllers in `toggleSideMenuHidden`:

```
// disable user interaction on the current stack of view controllers if the menu is visible
for(UIViewController* viewController in self.navigationController.viewControllers) {
    viewController.view.userInteractionEnabled = (self.menuState == MFSideMenuStateHidden);
}
```

That works fine for me. I've pushed this change to my fork of the repo. Will send a pull request later.
